crates/typst-cli/src/query.rs

```diff
@@ -0,0 +1,133 @@
+use comemo::Track;
+use serde::Serialize;
+use std::collections::HashMap;
+use typst::diag::{bail, StrResult};
+use typst::eval::Value::Dyn;
+use typst::eval::{eval_string, EvalMode, Tracer, Value};
+use typst::model::{Introspector, Selector};
+use typst::World;
+use typst_library::meta::ProvideElem;
+use typst_library::prelude::*;
+
+use crate::args::QueryCommand;
+use crate::compile::print_diagnostics;
+use crate::set_failed;
+use crate::world::SystemWorld;
+
+#[derive(Serialize)]
+pub struct SelectedElement {
+    #[serde(rename = "type")]
+    typename: String,
+    attributes: HashMap<EcoString, Value>,
+}
```

Comment:
I think this should be removed in favor of `Content::serialize`. `type` should be `func` and `attributes` should be `fields`.